2021.02 newbies are welcome (#501)
plone/training: mastering-plone/installation.rst

.. _installation-label:

Installation & Setup of **Plone 6**
=====================

**Wording**

A **Plone 6** installation is a combo of a Plone backend [1]_ with a Plone frontend [2]_.
All together we call it Plone.

.. [1] backend: your data is stored here.
.. [2] frontend: that's what your editors see and use.

| Whereas **Plone Classic** is still a valuable installation of Plone.
| *Plone Classic* is a valuable installation of a full featured CMS as it is since decades and still will be.
| *Plone Classic* means: A Plone installation for a Website. No ReactJS stuff, just Plone, just Python. Everything in one installation, backend and frontend.

**And now the sparkling Plone:**

| This training is about Plone.
| This training is about Plone 6.

.. _installation-plone-label:

Installing Plone backend
------------------------

Make sure you have a current and by Plone supported **Python 3** version. 
One way to achieve is `pyenv` which lets you manage different Python versions.
It even let's you setup virtual Pythons of the same version for individual projects.
https://github.com/pyenv/pyenv-installer

.. code-block:: console
    

>     pyenv install 3.8.10

Reviewer: ```suggestion
    pyenv install 3.9.5
```

>     pyenv virtualenv 3.8.10 plonepy

Reviewer: ```suggestion
    pyenv virtualenv 3.9.5 plonepy
```

    pyenv activate plonepy

> This installs and activates a Python 3.8.10. It does not affect your system Python as it is an isolated virtual Python environment.

Reviewer: ```suggestion
This installs and activates a Python 3.9.5. It does not affect your system Python as it is an isolated virtual Python environment.
```

Prerequisites
*************

The following instructions are based on Ubuntu and macOS.
If you use a different operating system (OS), please adjust them to fit your OS.

On Ubuntu/Debian, you need to make sure your system is up-to-date:

.. code-block:: console

    sudo apt-get update
    sudo apt-get -y upgrade

Then, you need to install the following packages:

.. code-block:: console

>     sudo apt-get install python3.7-dev python3.7-tk python3.7-venv build-essential libssl-dev libxml2-dev libxslt1-dev libbz2-dev libjpeg62-dev

Reviewer: For Plone 6 we shouldngonall the way with 3.9
